test/unittests/execute_test.cpp

```diff
@@ -1048,3 +1048,221 @@ TEST(execute, memory_copy_32bytes)
+      f64.const 1
+      f64.const 1
+      f64.lt
+      drop
+      f64.const 1
+      f64.const 1
+      f64.gt
+      drop
+      f64.const 1
+      f64.const 1
+      f64.le
+      drop
+      f64.const 1
+      f64.const 1
+      f64.ge
+      drop
+
+      ;; conversion ops
+      f64.const 1
+      f32.demote_f64
+      f64.promote_f32
+      drop
+
+      f32.const 1
+      i32.trunc_f32_s
+      drop
+      f32.const 1
+      i32.trunc_f32_u
+      drop
+      f64.const 1
+      i32.trunc_f64_s
+      drop
+      f64.const 1
+      i32.trunc_f64_u
+      drop
+
+      f32.const 1
+      i64.trunc_f32_s
+      drop
+      f32.const 1
+      i64.trunc_f32_u
+      drop
+      f64.const 1
+      i64.trunc_f64_s
+      drop
+      f64.const 1
+      i64.trunc_f64_u
+      drop
+
+      i32.const 1
+      f32.convert_i32_s
+      drop
+      i32.const 1
+      f32.convert_i32_u
+      drop
+      i64.const 1
+      f32.convert_i64_s
+      drop
+      i64.const 1
+      f32.convert_i64_u
+      drop
+
+      i32.const 1
+      f64.convert_i32_s
+      drop
+      i32.const 1
+      f64.convert_i32_u
+      drop
+      i64.const 1
+      f64.convert_i64_s
+      drop
+      i64.const 1
+      f64.convert_i64_u
+      drop
+
+      f32.const 1
+      i32.reinterpret_f32
+      drop
+      f64.const 1
+      i64.reinterpret_f64
+      drop
+      i32.const 1
+      f32.reinterpret_i32
+      drop
+      i64.const 1
+      f64.reinterpret_i64
+      drop
+    )
+    */
+
+    const auto bin = from_hex(
+        "0061736d0100000001060160027d7c000302010005030100010af50301f203004100430000803f38020041002a"
+        "020043c3f5484092430000803f9343000040409443000000409543000040409643000080409743000080bf988b"
+        "8c8d8e8f90911a430000803f430000803f5b1a430000803f430000803f5c1a430000803f430000803f5d1a4300"
+        "00803f430000803f5e1a430000803f430000803f5f1a430000803f430000803f601a410044000000000000f03f"
+        "39030041002b0300441f85eb51b81e0940a044000000000000f03fa1440000000000000840a244000000000000"
+        "0040a3440000000000000840a4440000000000001040a544000000000000f0bfa6999a9b9c9d9e9f1a44000000"
+        "000000f03f44000000000000f03f611a44000000000000f03f44000000000000f03f621a44000000000000f03f"
+        "44000000000000f03f631a44000000000000f03f44000000000000f03f641a44000000000000f03f4400000000"
+        "0000f03f651a44000000000000f03f44000000000000f03f661a44000000000000f03fb6bb1a430000803fa81a"
+        "430000803fa91a44000000000000f03faa1a44000000000000f03fab1a430000803fae1a430000803faf1a4400"
+        "0000000000f03fb01a44000000000000f03fb11a4101b21a4101b31a4201b41a4201b51a4101b71a4101b81a42"
+        "01b91a4201ba1a430000803fbc1a44000000000000f03fbd1a4101be1a4201bf1a0b");
+
+    const auto module = parse(bin);
+    auto instance = instantiate(module);
+
+    bool trap = false;
+    try
+    {
+        execute(instance, 0, {});
+    }
+    catch (unsupported_feature const&)
+    {
+        trap = true;
+    }
+    ASSERT_TRUE(trap);
```

Comment:
I think you could just use `ASSERT_THROW` instead of `try/catch`?

Reply:
`EXPECT_THROW_MESSAGE`?